[TS] LPP-47518 > LPS-171719 Handling the redirect in Action Process

ces-hongvo opened the pull request:
Hi @ces-binhtran 
This is the PR for the ticket https://issues.liferay.com/browse/LPP-47518. Please review it and leave your comments.
Thank you

### Motivation 
Redirect URL is a vital and vulnerable parameter. Therefore, leaving it in a form which can be manipulated is dangerous. This PR is addressing this issue
### Reproduce step:
Please see https://issues.liferay.com/browse/LPP-47518
### Solution Proposed
I will add a line with "_portal.escapeRedirect(...)" in the edit action

robertoDiaz committed:
LPS-163065 Allow configure google analytics 4 as legacy google analytics

robertoDiaz committed:
LPS-163065 Add lang keys

robertoDiaz committed:
LPS-163065 auto buildlang

jeffwu0724 committed:
LPS-171228 move joda-time and joda-convert to shared-dependencies

jeffwu0724 committed:
LPS-171228 add joda-time and joda-convert to source-formatter

jeffwu0724 committed:
LPS-171228 auto sf

SelenaAungst committed:
LPS-170337 Addd missing language key.

joelrocha68 committed:
LPS-172207 Removing Macros

joelrocha68 committed:
LPS-172207 Removing paths

joelrocha68 committed:
LPS-172207 Test Fix

jeffwu0724 committed:
LPS-171228 manually change compileInclude to compileOnly for joda-time and joda-convert

tinatian committed:
LPS-167572 SF, move logic from Scanner to DirectoryWatcher, prepare for next

tinatian committed:
LPS-167572 Apply FilenameFilter

tinatian committed:
LPS-167572 SF, move logic to DirectoryWatcher

tinatian committed:
LPS-167572 Use Canonical file consistently

tinatian committed:
LPS-167572 Initialize in constructor to avoid calculating at runtime

ccorreagg committed:
LPS-170791 Add filter parameter

ccorreagg committed:
LPS-170791 Move to osgi-service-tracker-collections package, remove interface and transform methods to static

ccorreagg committed:
LPS-170791 Update dependencies

ccorreagg committed:
LPS-170791 baseline

Commit:
LPS-170565 View object entries: The option to sort the column is disabled for Rich Text fields

jeffwu0724 committed:
LPS-171228 remove "!org.joda.time.*" in Import-Package because we need to imported it from shared-dependencies-joda. Add "org.joda.time.format" in Import-Package because third-party lib org.elasticsearch.elasticsearch has a dependency on it, we need to add it to make sure it can wire up

markocikos committed:
LPS-170299 Extend selection filter with autocomplete feature

markocikos committed:
LPS-170299 Apply standard preloaded data format

markocikos committed:
LPS-170299 Replace autocomplete filter with selection filter. Apply standard preloaded data format.

markocikos committed:
LPS-170299 Update semver

fmarcio committed:
LRAC-12624 set const to be exported in App component

fmarcio committed:
LRAC-12624 add prop on Attributes component

fmarcio committed:
LRAC-12624 Create unit tests

fmarcio committed:
LRAC-12624 update TS file

fmarcio committed:
LRAC-12624 Create unit tests TS files

ealonso committed:
LPS-171687 Removed no needed files

ealonso committed:
LPS-171687 Implements item selector default view

ealonso committed:
LPS-171687 It is not necessary to use getUnambiguousTitle, since we are showing entries in the same group

ealonso committed:
LPS-171687 Implements title and subtitle

Commit:
LRAC-12639 - Create unit  tests for Attributes Page

Commit:
LRAC-12639 - Create unit  tests for Connect Page

Commit:
LRAC-12639 - Create unit  tests for People Page

Commit:
LRAC-12639 - Create unit  tests for Properties Page

Commit:
LRAC-12639 - Create unit  tests for Default Page

Commit:
LRAC-12639 - TD generated files

ealonso committed:
LPS-171687 Update html using taglib parameters

ealonso committed:
LPS-171687 build.gradle

ealonso committed:
LPS-171687 Adds new interface to represent a table view in item selector

ealonso committed:
LPS-171687 Implements DefaultTableItemView using the ItemDescriptor

ealonso committed:
LPS-171687 Uses new class and interface to render the table view

ealonso committed:
LPS-171687 Adds new method to return the TableItemView

ealonso committed:
LPS-171687 Implements table custom item view for asset list entries

ealonso committed:
LPS-171687 Make name selectable

ealonso committed:
LPS-171687 baseline

ealonso committed:
LPS-171687 Removes descriptive view for the moment

shuyangzhou committed:
LPS-172185 Remove non-exist 136 portlet

shuyangzhou committed:
LPS-172185 No longer needs portal ServletContext

shuyangzhou committed:
LPS-172185 Apply usages

shuyangzhou committed:
LPS-172185 Regenerate

JanisZhang committed:
LPS-171374 Remove Optional usages in AccountResourceImpl

JanisZhang committed:
LPS-171374 Remove Optional usages in DDMFormValuesUtil

JanisZhang committed:
LPS-171374 Remove Optional usages in FormResourceImpl

JanisZhang committed:
LPS-171374 Remove Optional usages in PageElementDTOConverter

JanisZhang committed:
LPS-171374 Remove Optional usages in KnowledgeBaseArticleDTOConverter

JanisZhang committed:
LPS-171374 Remove Optional usages in WikiPageDTOConverter

shuyangzhou committed:
LPS-172078 Remove org.apache.aries.jax.rs.jackson to avoid its OSGi DSL usage

marcelabc committed:
LPS-170299 Sort

luisacmn committed:
LPS-171605 Request, filter data and populate chart

luisacmn committed:
LPS-171605 SF

julschong committed:
LPS-171810 Auto, gw -b releng.gradle cleanReleng, cleanup portal-template-soy-renderer artifact and changelog

tinatian committed:
LPS-167572 Add integration test to expose the issue

jeffwu0724 committed:
LPS-171857 remove Stream usages in saml-api and use TransformUtil instead

jeffwu0724 committed:
LPS-171857 remove the use of Optional in saml-api

JanisZhang committed:
LPS-171374 Remove Optional usages in PageDefinitionDTOConverter

JanisZhang committed:
LPS-171374 Remove Optional and Stream usages in StructuredContentResourceImpl

ces-quanhuynh committed:
LPS-171824 Add language key reminder-query-answer

ces-quanhuynh committed:
LPS-171824 Build lang

JanisZhang committed:
LPS-171374 Remove Stream usages in CustomFieldsUtil

JanisZhang committed:
LPS-171374 Remove Optional usages in PortalInstanceResourceImpl

robertoDiaz committed:
LPS-163065 Update variable name

jeffwu0724 committed:
LPS-171857 add dependencies on petra-function to use TransformUtil in saml-api

luisacmn committed:
LPS-171605 Refactoring

JhossephG committed:
LPS-171799 Add CanCreateActionNotificationWithOnPaymentStatusUpdate Test

fortunatomaldonado committed:
LPS-170337 Ran buildLang

luisacmn committed:
LPS-171605 Refactoring: Add missing error msg

Commit:
LPS-171554 Create Claims Settled Chart and Filter

michaelhashimoto committed:
LRCI-3371 Add ability to see if the required suites are completed in Pull Requests

joshitagaki committed:
LRAC-12693 Test fix dropdownIndex is set

holatuwol commented:
There used to be a source formatting rule that ensures we call `PortalUtil.escapeRedirect` for redirect parameters, so it looks like it was either removed or it stopped working.

Can you open a PTR with Application Security and ask them what happened to it, and if it still exists, if they have any idea why this one was missed?

JanisZhang committed:
LPS-171374 SF

22LucaGiannotta22 committed:
COMMERCE-10548 Skip the click if there's only one vocabulary

Commit:
LPS-171434 Fixed MDF Dashlet doesn't support long characters

sandrodw3 committed:
LPS-171896 Adjust Autosave icon so device icons are not moved

jeffwu0724 committed:
LPS-171228 auto generate, by running "ant build-lib-versions"

DiegoHu97 committed:
LPS-171896 Add language key

DiegoHu97 committed:
LPS-171896 buildLang

devpedrotavares committed:
LPS-171610 Fix bug by redirecting

Commit:
LPS-171918 Fix RESULT_RANKINGS_ALIAS_VALUE path

Commit:
LPS-171434 Simplify low code components inside Guest page

Commit:
LPS-171752 Return the correct name

sandrodw3 committed:
LPS-171896 Use a ref

sandrodw3 committed:
LPS-171896 Improve accesibility

DiegoHu97 committed:
LPS-171022 The dropdown menu of category selector overflows the window

sandrodw3 committed:
LPS-169588 This is no needed, we want secondary default color

Jeremy-Ch committed:
LPS-171687 Update macros to use the suitable path

ealonso committed:
LPS-165330 Convert into a string since when we are deleting we are doing:

const index = instance._ids.indexOf(id.toString());

BarbaraCabrera committed:
LPS-171952 Replace liferay-ui:icon with clay:link in view.jsp

lfbesada committed:
LPS-171399 Don't assume that the main field is the first one, instead search it by name.

When a compatibility fieldSet is added to handle the old nested fields, the name of the parent field is used, adding "FieldSet" as suffix. In addition, this parent field is set as the first child of the compatibility FieldSet, followed by its nested fields, so it result as the first sibling of its old nested fields.

See: https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/content/compatibility/converter/JournalContentCompatibilityConverterImpl.java#L282-L303

But some changes in the data definition could change the nested fields sort resulting that the old parent field won't be the first child anymore.

lfbesada committed:
LPS-171399 Don't use the index to remove it, use the TemplateNode

lfbesada committed:
LPS-171399 SF rename

lfbesada committed:
LPS-171399 Improve the performance, don't include backward compatibility nodes for the new versions nested fields.

If the FieldSet name doesn't end with "FieldSet" then it isn't an added compatibility FieldSet, so we don't need to include a compatibility node.

Its child nodes won't need a compatibility TemplateNode either because of their parent is a new FieldSet, so they can't be an old style nested fields.

lfbesada committed:
LPS-171399 Improve the performance, don't add useless nodes. If the mainChildTemplateNode could be found by name, is useless to add a compatibility TemplateNode with other name, so we won't do.

But we are going to process all the child nodes because some of them could have an old style nested fields.

victorg1991 committed:
LPS-169588 This is also no needed

GaborKomaromi committed:
LPS-171955 only generate sitemaps for active groups

ealonso committed:
LPS-171973 Removes unused methods

victorg1991 committed:
LPS-165330 Sent the value directly in string since long values cannot be represented well in javascript

victorg1991 committed:
LPS-165330 Not needed

ambrinchaudhary committed:
LPS-171740 Dropdown actions must be normalized, not the items

ambrinchaudhary committed:
LPS-171740 Add aria-label to solve accessibility issues

ambrinchaudhary committed:
LPS-171740 Update test

ealonso committed:
LPS-171973 Simplify

ealonso committed:
LPS-171973 baseline

victorg1991 committed:
LPS-171668 Reset Display Date values when editing default values

ealonso committed:
LPS-171973 Revert changes in LRQA-60221

BarbaraCabrera committed:
LPS-171953 Replace liferay-ui:icon with clay:link in view_container_model.jsp

victorg1991 committed:
LPS-171969 Remove ff for LPS-166201

victorg1991 committed:
LPS-171969 Not needed

victorg1991 committed:
LPS-171969 Remove ff for LPS-166203

ealonso committed:
LPS-171973 Uses constant since we are using and old constant that is no longer supported

ealonso committed:
LPS-171973 We should set thread local if no, we are going to create a new version

ealonso committed:
LPS-171973 Fix: it should be rich text

ealonso committed:
LPS-171973 build.gradle

ealonso committed:
LPS-171973 Removes last new line

ealonso committed:
LPS-171973 Don't compare strings with !=

BarbaraCabrera committed:
LPS-171971 Replace liferay-ui:icon-help with clay:icon in edit_asset_vocabulary.jsp

JanisZhang committed:
LPS-170769 SF

Commit:
LPS-171896 Modify locator to adapt new markup

slnn committed:
LPS-171992 Remove the usage of Optional as return value in OAuth2Manager methods

BarbaraCabrera committed:
LPS-171974 Replace all Liferay-ui:icon for clay:icon in view.jsp

slnn committed:
LPS-171992 Remove the usage of Option as return value in AccessTokenStoreUtil.getAccessTokenOptional(long companyId, long userId) and rename it to getAccessToken(long companyId, long userId)

lfbesada committed:
LPS-172000 Take into account that the list size could be minor than end parameter

lfbesada committed:
LPS-172000 Ensure that the start is always minor than the end otherwise, return an empty collection

slnn committed:
LPS-171992 Remove the usage of Optional as return value in OAuth2StateUtil.getOAuth2StateOptional(HttpServletRequest httpServletRequest) and rename to getOAuth2State(HttpServletRequest httpServletRequest)

slnn committed:
LPS-171992 Apply

slnn committed:
LPS-171992 Apply

slnn committed:
LPS-171992 Apply

slnn committed:
LPS-170480 Remove the usage of Option as return value in BaseAssetDisplayPageFriendlyURLResolver(String template, InfoItemFieldValues infoItemFieldValues, Locale locale)

slnn committed:
LPS-170480 Apply

BarbaraCabrera committed:
LPS-172006 Replace liferay-ui:icon in asset_selection_action.jsp

BarbaraCabrera committed:
LPS-172007 Replace all liferay-ui:icon with clay:link in asset_selection_order_down_action.jsp

BarbaraCabrera committed:
LPS-172008 Replace all liferay-ui:icon with clay:link in asset_selection_order_up_action.jsp

victorg1991 committed:
LPS-171986 Fix condition and pass prop

BarbaraCabrera committed:
LPS-172007 SF, simplify to avoid duplication

carolmariaabb committed:
LPS-161287 Only overwrite user input if value is not null or not empty

rafaellajordao committed:
LPS-165992 Automate ObjectFields#CanNotOrderByAggregationFieldColumn

ealonso committed:
LPS-172013 It should be is null

rafaellajordao committed:
LPS-162194 Automate ObjectFields#CanCreateActionWithCustomFieldInSystemObjects

rafaellajordao committed:
LPS-165981 Automate ObjectFields#ExcludesIsAvailableForPicklistFields

rafaellajordao committed:
LPS-165980 Automate ObjectFields#IncludesIsAvailableForPicklistFields

fmarcio committed:
LRAC-12624 SF

luisacmn committed:
LPS-171605 Refactoring: Simplify code

BarbaraCabrera committed:
LPS-172010 Replace all liferay-ui:icon with clay:button in ordering.jsp

Commit:
LPS-171630 Refactor Button.clickSave to PortletEntry.save

Commit:
LPS-171630 Add CanEditEntryWhenAllowedToEdit test

luisacmn committed:
LPS-171605 Refactoring: Added getChartValues file

BarbaraCabrera committed:
LPS-172011 Replace liferay-ui:icon with clay:icon in scope.jsp

Commit:
LPS-171590 Refactor CanAddEntryOnRelationshipTabFor1toM test

luisacmn committed:
LPS-171605 Refact: Add getDeals functions files

luisacmn committed:
LPS-171605 SF

Commit:
LPS-171597 Refactor addEntryOnRelationshipTab macro

Commit:
LPS-171597 Add CanAddEntryOnRelationshipTabForMtoM test

biellopes00 committed:
LPS-172021 Change name of fieldKey on page-definition to show the name of company on detail and review page

kiwm committed:
LPS-171635 Change the getMaxDateActivity.ts and getMinDateActivity.js logic

lucasmiranda0 committed:
LPS-171483 Creating options to set timeZone to UTC

michaelhashimoto committed:
LRCI-3385 Change Testray Build IDs to longs instead of ints

tinatian committed:
LPS-171752 Fix typo, it is collecting disabled features

Commit:
LPS-171918 Fix RESULT_RANKINGS_ALIAS_REMOVE Path

Commit:
LPS-171632 Add CanDeleteNewEntryAddedOnRelationshipTab test

Brunno0liveira committed:
COMMERCE-10542 Maximum quantity can be added to cart & Maximum quantity cannot be added to cart

Commit:
LPS-171632 Refactor Button.clickSave to PortletEntry.save

michaelhashimoto committed:
LRCI-3385 Change Testray Product Version IDs to longs instead of ints

michaelhashimoto committed:
LRCI-3385 Change Testray Routine IDs to longs instead of ints

michaelhashimoto committed:
LRCI-3385 Change Testray Project IDs to longs instead of ints

Commit:
LPS-171629 Add CanCancelNewEntryOnRelationshipTab test

michaelhashimoto committed:
LRCI-3385 Change item id to a long instead of an int

Commit:
LPS-171628 Add CanSelectExistingEntriesOnRelationshipTab test

drewbrokke committed:
LPS-171893 change-tracking-service: always remove the collection id so the preview id will be read

brookedalton committed:
LPS-171918 Update paths

tinatian committed:
LPS-167572 Move logic to util, prepare for next

Commit:
LPS-171969 Refactor macro to adapt new behavior

Commit:
LPS-171969 Use macro to simplify case

Commit:
LPS-171969 Refactor case to adapt new behavior

Commit:
LPS-171969 Remove deprecated steps

Commit:
LPS-171969 Update usages

dantewang committed:
LPS-171992 SF, proper assert

Jeremy-Ch committed:
LPS-170168 Add macros for assigning the depot role to user

Jeremy-Ch committed:
LPS-170168 Use JSON to add user to depot

Jeremy-Ch committed:
LPS-170168 Use JSON to add depot role to user

Jeremy-Ch committed:
LPS-170168 Add macros for add depot role by JSON

Jeremy-Ch committed:
LPS-170168 Use JSON to add depot role

Jeremy-Ch committed:
LPS-170168 Rename the tests

dsanz committed:
COMMERCE-10546 quick fix: prevent ClayPaginationBarWithBasicItems to reneder if there are no items

Jeremy-Ch committed:
LPS-170168 Use JSON to edit user info

victorg1991 committed:
LPS-172007 Merge conditions

victorg1991 committed:
LPS-172007 Use aria-label and title directly on the clay:link

lfbesada committed:
LPS-171399 Adds asserts to make the test fail if the result doesn't match the expectations instead of throwing an exception

lfbesada committed:
LPS-171399 Uses the proper field names.

When a compatibility fieldSet is added to handle the old nested fields, the name of the parent field is used, adding "FieldSet" as the suffix. So if we want to test the backward compatibility we have to use the TemplateNodes with names matching the backward logic.

lfbesada committed:
LPS-171399 SF rename

lfbesada committed:
LPS-171399 SF remove unused

victorg1991 committed:
LPS-172011 It seems that this code is not used, so remove it

lfbesada committed:
LPS-171399 The fields nested in a fieldSet need to be accessed throw its fieldSet.

It was working properly because we were adding backward compatibility nodes to the newly produced data.

balazssk committed:
LPS-172129 Fix the order of parameters

lfbesada committed:
LPS-171399 Sets dynamically the fieldSet name

lfbesada committed:
LPS-171399 Adds a test to ensure that when the fieldSet field name matches the name of a backward compatibility added one then the backward compatibility nodes will be added to the Freemarker context.

In that way, the nested field will also be accessible in the template in the same way as in older versions.

victorg1991 committed:
LPS-172011 Use taglib attributes

interaminense committed:
LRAC-12711 Just expand first column on tables

veroglez committed:
LPS-171971 Include the radio buttons inside a group and read the help text with screen reader

veroglez committed:
LPS-171974 Replace code by clay:alert

kiwm committed:
LPS-171635 Change the getMaxDateActivity.ts and getMinDateActivity.js logic updated

Co-authored-by: Thaynam Lázaro <lazaro@thaynam.com>

Commit:
LPS-171631 Add CannotEditEntryWhenNotAllowedToEdit test

Commit:
LPS-171630 Small corrections for better readability

michaelhashimoto committed:
LRCI-3217 Add operating.system.types as a poshi property

tinatian committed:
LPS-167572 Only process configuration files in configuration folder

michaelhashimoto committed:
LRCI-3217 Add operating system types to all the environment poshi tests

luisacmn committed:
LPS-171605 Refactoring: reduce approach

marcosapmf committed:
LRAC-12710 enable sort by field type

luisacmn committed:
LPS-171605 Add useMemo

leilanyaragao committed:
LRAC-12707 add userName in contact field

luisacmn committed:
LPS-171605 Removed unnecessary utils files

marcosapmf committed:
LRAC-12709 add missing field

marcosapmf committed:
LRAC-12709 SF

brookedalton committed:
LPS-172074 Set http.host if browser docker image is set

Commit:
LPS-172180 Fix ConfigureDisplayTimeZone test

Commit:
LPS-171904 Fix GuestCanAddObjectEntriesWithAddPermission test

shuyangzhou committed:
LPS-172185 Semantic versioning

joshchong committed:
LPS-172187 Unquarantine Synonyms integration tests from search and relevant suites

dantewang committed:
LPS-171857 Sem Ver

lfbesada committed:
LPS-172146 Use the localized pattern

Songyuewen committed:
LRQA-78315 Add test view column selected icon

sirfabios committed:
LPS-171802 Add SECTION_TITLE xpath

sirfabios committed:
LPS-171802 Add ObjectActions#VerifyTheConditionCardAreHidden test

Brunno0liveira committed:
COMMERCE-10542 Auto SF

brianchandotcom committed:
LPS-171374 Rename

brianchandotcom committed:
LPS-171374 SF

Commit:
LPS-169235 Fix CanOrderByResource test

brianchandotcom committed:
LRCI-3385 prep next

brianchandotcom committed:
LPS-171955 permission checks not needed here

brianchandotcom committed:
LPS-170299 SF

brianchandotcom committed:
LPS-170299 baseline

brianchandotcom committed:
LPS-167572 Wordsmith

brianchandotcom committed:
LPS-171896 Wordsmith

brianchandotcom committed:
LPS-171896 Regen

brianchandotcom committed:
artifact:ignore journal-service 7.0.102 prep next

liferay-translation committed:
LPS-77699 Update Translations

ces-hongvo committed:
LPS-171719 Escape Redirect URL in PasswordPoliciesAdminPortlet

hongvo2308 commented:
Forwarded https://github.com/liferay-appsec/liferay-portal/pull/961
